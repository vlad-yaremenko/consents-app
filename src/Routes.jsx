import React from 'react';
import {
  Switch,
  Route,
  Redirect
} from "react-router-dom";

// Routes
import GiveConsent from '@views/GiveConsent';
import Consents from '@views/Consents';

const Component = () => (
  <Switch>
    <Route exact path="/">
      <Redirect to="/give-consent"/>
    </Route>
    <Route path="/give-consent">
      <GiveConsent />
    </Route>
    <Route path="/consents">
      <Consents />
    </Route>
  </Switch>
);
Component.displayName = 'Routes';

export default Component;
