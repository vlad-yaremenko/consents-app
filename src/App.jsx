import './App.scss';
import DefaultLayout from '@layouts/Default';
import Routes from './Routes';

const Component = () => (
  <DefaultLayout>
    <Routes/>
  </DefaultLayout>
);
 Component.displayName = 'App';

export default Component;
