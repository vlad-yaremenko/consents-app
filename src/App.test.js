import { render } from '@testing-library/react';
import App from './App';
import DefaultLayout from '@layouts/Default';
import Routes from './Routes';

jest.mock('@layouts/Default', () => {
  return jest.fn()
});
jest.mock('./Routes', () => {
  return jest.fn();
})

beforeEach(() => {
  DefaultLayout.mockImplementation(({ children }) => <div data-testid="default-layout">{children}</div>);
  Routes.mockImplementation(() => <p data-testid="routes">Routes</p>);
});

afterEach(() => {
  jest.resetAllMocks();
});

test('should render DefaultLayout with routes', () => {
  const { container, getByTestId } = render(<App />);

  expect(getByTestId('default-layout')).toBeInTheDocument();
  expect(getByTestId('routes')).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
