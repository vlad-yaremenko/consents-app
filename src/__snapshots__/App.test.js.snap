// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DefaultLayout with routes 1`] = `
<div>
  <div
    data-testid="default-layout"
  >
    <p
      data-testid="routes"
    >
      Routes
    </p>
  </div>
</div>
`;
